docker-compose.yml

```
@@ -484,6 +485,30 @@ services:
     volumes: *ubuntu-volumes
     command: *cpp-command
 
+  ubuntu-cpp-bundled-offline:
+    # Arrow build with BUNDLED dependencies with downloaded dependencies.
+    image: ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-minimal
+    build:
+      context: .
+      dockerfile: ci/docker/ubuntu-${UBUNTU}-cpp-minimal.dockerfile
+      cache_from:
+        - ${REPO}:${ARCH}-ubuntu-${UBUNTU}-cpp-minimal
+      args:
+        arch: ${ARCH}
+        base: "${ARCH}/ubuntu:${UBUNTU}"
+        llvm: ${LLVM}
+    shm_size: *shm-size
+    ulimits: *ulimits
+    environment:
+      <<: [*common, *ccache, *sccache, *cpp]
```

I will admit that my knowledge is slightly sketchy here, but do we want ccache and sccache here? Is it possible those could contain built artifacts that aren't offline?

---

Good point.
ccache can be used for local `archery docker run`.
sccache will not work...

Let's remove sccache.

```suggestion
      <<: [*common, *ccache, *cpp]
```